packages/ui/src/app/pages/WorkingGroups/MyApplications.tsx

```diff
@@ -18,20 +19,35 @@ export const MyApplications = () => {
     ],
     []
   )
-  const { applications } = useMyApplications()
+  const { applications, isLoading } = useMyApplications()
+  const isPending = (a: WorkingGroupApplication) => a.status == 'ApplicationStatusPending'
+  const currentApplications = useMemo(() => applications?.filter(isPending), [applications])
+  const pastApplications = useMemo(() => applications?.filter((a) => !isPending(a)), [applications])
+  const pickText = () => {
+    if (isLoading) return 'Loading...'
+    return applications?.length ? 'My Applications' : 'No applications found'
+  }
 
   return (
     <AppPage crumbs={crumbs}>
       <PageHeader>
         <PageTitle>Working Groups</PageTitle>
         <WorkingGroupsTabs />
       </PageHeader>
-      <TextBig>My Applications</TextBig>
+      <TextBig>{pickText()}</TextBig>
       <MainPanel>
-        <Label>CURRENT APPLICATIONS</Label>
-        <ApplicationsList applications={applications?.filter((a) => a.status == 'ApplicationStatusPending') ?? []} />
-        <Label>PAST APPLICATIONS</Label>
-        <ApplicationsList applications={applications?.filter((a) => a.status != 'ApplicationStatusPending') ?? []} />
+        {currentApplications?.length ? (
+          <>
+            <Label>Current applications</Label>
+            <ApplicationsList applications={currentApplications} />
+          </>
+        ) : null}
+        {pastApplications?.length ? (
+          <>
+            <Label>Past applications</Label>
+            <ApplicationsList applications={pastApplications} />
+          </>
+        ) : null}
```

Comment:
```suggestion
        {pastApplications?.length && (
          <>
            <Label>Past applications</Label>
            <ApplicationsList applications={pastApplications} />
          </>
        ) }
```

should be shorter, but I don't recall if that would work (guessing on the `null` use).